tasks/main.yml

```diff
@@ -40,16 +40,60 @@
       list | length > 0 }}"
     __firewall_python_cmd: "{{ ansible_python_interpreter |
       d(discovered_interpreter_python) }}"
+    __firewall_report_changed: true
 
-- name: Get config files, checksums before and remove
-  script:
-    cmd: >
-      files/get_files_checksums.sh {{ __firewall_python_cmd | quote }}
-      {{ __firewall_firewalld_dir | quote }} true
-  register: __firewall_config_files_before
-  changed_when: false
+- name: Handle previous replaced
   when: __firewall_previous_replaced | bool
+  block:
+    - name: Get config files, checksums before and remove
+      script:
+        cmd: >
+          files/get_files_checksums.sh {{ __firewall_python_cmd | quote }}
+          {{ __firewall_firewalld_dir | quote }} true
+          {{ __firewall_package_with_conf }}
+      register: __firewall_config_files_before
+      changed_when: false
+      when: not ansible_check_mode
```

Comment:
I think the ansible.builtin.script will refuse to run in check mode anyways without `check_mode: false` or either `removes` or `creates` set with the error msg:
"Check mode is not supported for this task."

https://github.com/ansible/ansible/blob/010e542a13516eed9bb6b5e997ea255104257a80/lib/ansible/plugins/action/script.py#L100

Keeping `not ansible_check_mode` Is  probably the best choice, since we cannot guarantee that ansible will keep this behavior in later releases.

Reply:
I rewrote this somewhat